### ISSUE TITLE
Exception Call to a member function get_tasks() on null

### ISSUE THREAD

**Issue by truesh324** [donhinkelman/moodle-block_sharing_cart#101]
We are getting the following error with sharing cart in Moodle 3.10

I think this is similar to other issues.
https://github.com/donhinkelman/moodle-block_sharing_cart/issues/97

Shawn Trueman
Moodle Admin - Denver Seminary

**Comment by truesh324** [comment]
![sharing cart error](https://user-images.githubusercontent.com/31080814/123857634-ccc98e80-d8df-11eb-9b4f-4a232e0a5b0b.png)

**Comment by donhinkelman** [comment]
Thank you, Shawn, for reporting this error message. Could you confirm that this occured with the latest release of Sharing Cart 3.10 release 3, either from Github or Plugins Database?

**Comment by truesh324** [comment]
Don – We do have newest version.
***@***.***D76E64.B63F5DB0]

Our host, Elearning Experts will be completely uninstalling and reinstalling the plugin today to see if the error might correct.



Your servant,

Shawn Trueman, M.A., M.M., LPCC

Director of Educational Technology
Denver Seminary
6399 South Santa Fe Drive, Littleton, CO 80120
(O) 303.762.6959            (C) 817.995.6760

Please note my regular office hours:
Monday: 7a-4p
Tuesday: 8a-5p
Wednesday: 8a-5p
Thursday: 8a-5p
Friday: Out of Office
Saturday – Technical Support as Needed
~ Strengths Finder: Belief, Achiever, Responsibility, Harmony, Relator

Christians desire more theological training, but they struggle to fit it into their demanding lives. We offer accessible, quality, fully-online programs<https://denverseminary.edu/academics/fully-online-degrees/>, so they can become equipped for service and transformed by God.

**Comment by abias** [comment]
Running Moodle core 3.9.8, I was seeing this rhis fatal error with block_sharing_cart 3.10, release 3 for some days now.
After some investigation, I found that the reason was a local configuration change some days ago which let the plugin fail.

I have submitted a fix in #106 and assume that this PR will solve this ticket here as well.

**Comment by truesh324** [comment]
Greetings - 
Wondering if there is a timeline for the correction of this issue. I would like to be able to communicate with our users as to when we will be up and running. We use sharing cart often for exams and we are beginning to prep for mid terms.
Thanks.

**Comment by donhinkelman** [comment]
Alex fixed this issue and I merged the pull request. So it should have be solved in Github.  Are you downloading the block from GitHub or Moodle Plugins database?

Don

**Comment by abias** [comment]
@truesh324 ,

I was seeing basically the same error on my site, tracked it down, saw a problem with sites which use the $CFG->backuptempdir configuration and submitted #106 to fix this.

This issue here is still open and would need some confirmation from you if #106 fixed the issue or if some other problem causes this on your site.

However, as @donhinkelman wrote, this fix isn't published officially on https://moodle.org/plugins/block_sharing_cart/versions yet. The official versions are somehow lagging behind currently. Please test the latest version here on Github to see if this issue is fixed for you or not.

If it is fixed, this issue can be closed.
And if you can't use the Github version directly, you can apply the patch from https://github.com/donhinkelman/moodle-block_sharing_cart/pull/106/files on your site for the time being.

Cheers,
Alex

**Comment by truesh324** [comment]
Thanks Alex and Don for this information.
I will try to download from github, but I’ve not done that before.
Is this the process?

  1.  Click the green “code” button
  2.  Select download ZIP
  3.  Save file “moodle-block_sharing_cart-master”
  4.  Then unzip and rename to “moodle-block_sharing_cart”
  5.  Upload to Moodle

Your servant,

Shawn Trueman, M.A., M.M., LPCC

Director of Educational Technology
Denver Seminary
6399 South Santa Fe Drive, Littleton, CO 80120
(O) 303.762.6959            (C) 817.995.6760

Please note my regular office hours:
Monday: 7a-4p
Tuesday: 8a-5p
Wednesday: 8a-5p
Thursday: 8a-5p
Friday: Out of Office
Saturday – Technical Support as Needed
~ Strengths Finder: Belief, Achiever, Responsibility, Harmony, Relator

Christians desire more theological training, but they struggle to fit it into their demanding lives. We offer accessible, quality, fully-online programs<https://denverseminary.edu/academics/fully-online-degrees/>, so they can become equipped for service and transformed by God.

**Comment by abias** [comment]
Hi @truesh324 ,

I would rephrase your steps 4 and 5 to:

  4.  Then unzip the downloaded ZIP file and rename the resulting folder to “sharing_cart”
  5.  Place the folder into the "blocks" directory of Moodle

Apart from that, you got the general process right. Good luck!

Cheers,
Alex

**Comment by truesh324** [comment]
Thanks so much for your work on this plugin.

It is installed and functioning properly now for us.


Your servant,

Shawn Trueman, M.A., M.M., LPCC

Director of Educational Technology
Denver Seminary
6399 South Santa Fe Drive, Littleton, CO 80120
(O) 303.762.6959            (C) 817.995.6760

Please note my regular office hours:
Monday: 7a-4p
Tuesday: 8a-5p
Wednesday: 8a-5p
Thursday: 8a-5p
Friday: Out of Office
Saturday – Technical Support as Needed
~ Strengths Finder: Belief, Achiever, Responsibility, Harmony, Relator

Christians desire more theological training, but they struggle to fit it into their demanding lives. We offer accessible, quality, fully-online programs<https://denverseminary.edu/academics/fully-online-degrees/>, so they can become equipped for service and transformed by God.

**Comment by abias** [comment]
Thank you for confirming that the latest version fixed this issue for you.

@donhinkelman , regardless of the fact that an official release on https://moodle.org/plugins/block_sharing_cart/versions with the latest fixes is still pending, this issue could then be closed.

**Comment by donhinkelman** [comment]
To Shawn, (>> It is installed and functioning properly now for us.). Thank you for doing this testing, because I did not have the time to properly test it, and now I feel confident to make an official release on the Moodle Plugins database. Your contribution helps make Moodle grow.  And thank you, Alex for coaching Shawn as we have to help each other in so many ways. I will close this issue and proceed with a new release.
